### DISCUSSION TITLE
How to subscribe my Next.js service worker to push notifications

### DISCUSSION THREAD

**Discussion by FleetAdmiralJakob** [serwist/serwist#153] [Q&A] [answered]
I currently have set up the default serwist next.js stuff and want to subscribe my service worker to push notifications. How can I do that with serwist?

I mean the Push API: https://developer.mozilla.org/en-US/docs/Web/API/PushManager/subscribe

**Comment by DuCanhGH** [comment, marked as answer]
@FleetAdmiralJakob see [our example here](https://github.com/serwist/serwist/tree/main/examples/next-web-push).

**Reply by FleetAdmiralJakob** [reply]
Oh, great. I parsed whole GH for an example but did not see this one. Thank you!

**Reply by FleetAdmiralJakob** [reply]
Hi, I implemented the stuff from the example into my app, but I have one problem left.

If I close the Chrome PWA on Windows, the push notifications do no longer get sent.


1. Is that an issue with my implementation, or does the example also not have this feature?

2. If it's an issue inside the example: How can I achieve that, Windows does get them even if the browser is closed?

**Reply by DuCanhGH** [reply]
@FleetAdmiralJakob that seems weird. Notifications should work even without the PWA opened.

**Reply by FleetAdmiralJakob** [reply]
Oh, I think it's a Chrome on Windows problem.

The problem is that on Edge I get the notification cause it's a windows endpoint too, but on chrome on windows I get a google endpoint (fcm.googleapis.com) and that does not send the notification if chrome is closed.

That's sooo broken. Can you verify the issue on chrome, and that the issue is not on edge, so I can make a bug report to the chrome team?

I made a minimal reproduction based on your example. You can put the subscription object in the sendNotification file and then execute it via pnpm and tsx so you can send messages even if chrome/edge is closed.

Repro: https://github.com/FleetAdmiralJakob/serwist-next-push-api-test

**Reply by DuCanhGH** [reply]
@FleetAdmiralJakob I, well, do not use Chrome and Windows, so you should just go ahead and open a new issue if that's actually the case. Sounds weird that no one has noticed such a bug if it actually exists though.

Edit: according to section 2 of https://caniuse.com/push-api, desktop Chrome, Firefox, and Edge all require the browser to be running for you to receive messages. This doesn't appear to be the case for mobile ones, though.

**Reply by FleetAdmiralJakob** [reply]
But then canius is based on my testing wrong with this one, since I can receive messages via edge even if edge is closed. I will have to file the issue there then.

**Reply by DuCanhGH** [reply]
@FleetAdmiralJakob maybe, just maybe, Edge wasn't fully closed when you were testing that? Seems that certain plugins and features may keep the browser running even when there are no opened windows. Well, you know, since Edge is that integrated into the OS and all.
